hypothesis-python/src/hypothesis/core.py

```diff
@@ -280,12 +276,6 @@ def wrapped_test(*arguments, **kwargs):
             "%s() got an unexpected keyword argument %r, from `%s=%r` in @given"
             % (name, arg, arg, given_kwargs[arg])
         )
-    for a in original_argspec.args:
-        if isinstance(a, list):  # pragma: no cover
-            return invalid(
-                "Cannot decorate function %s() because it has destructuring arguments"
-                % (name,)
-            )
```

Comment:
Why was this bit removed? Is this a change in how Python 3 handles argspec?

Reply:
```python
def fxn(a, (b, c), d):
    pass
```

was made invalid in Python 3 by https://www.python.org/dev/peps/pep-3113/